image-segmentation/custom-recipes/score_image/recipe.py

```diff
@@ -0,0 +1,140 @@
+DETAILS_DIR = details_folder.get_path()
+# -------------------------------------------------------------------------------- NOTEBOOK-CELL: CODE
+class InferenceConfig(coco.CocoConfig):
+    # Set batch size to 1 since we'll be running inference on
+    # one image at a time. Batch size = GPU_COUNT * IMAGES_PER_GPU
+    GPU_COUNT = 1
+    IMAGES_PER_GPU = len(image_folder.list_paths_in_partition(partition=''))
+
+config = InferenceConfig()
+
+model = modellib.MaskRCNN(mode="inference", model_dir=model_folder_path, config=config)
+model.load_weights(COCO_MODEL_PATH, by_name=True)
+
+
+class_names = ['BG', 'person', 'bicycle', 'car', 'motorcycle', 'airplane',
+               'bus', 'train', 'truck', 'boat', 'traffic light',
+               'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird',
+               'cat', 'dog', 'horse', 'sheep', 'cow', 'elephant', 'bear',
+               'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie',
+               'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball',
+               'kite', 'baseball bat', 'baseball glove', 'skateboard',
+               'surfboard', 'tennis racket', 'bottle', 'wine glass', 'cup',
+               'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
+               'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza',
+               'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed',
+               'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote',
+               'keyboard', 'cell phone', 'microwave', 'oven', 'toaster',
+               'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors',
+               'teddy bear', 'hair drier', 'toothbrush']
+
+# -------------------------------------------------------------------------------- NOTEBOOK-CELL: CODE
+to_score = image_folder.list_paths_in_partition(partition='')
+image_list=[]
+for im in to_score :
+    image_path = IMAGE_DIR+im
+    image = skimage.io.imread(image_path)
+    image_list.append(image)
+
+result=model.detect(image_list, verbose=1)
+
+for i, im in enumerate(result):
+    result[i]["image_path"] = to_score[i]
+result_json=[]
+
+def jsonify(data):
+    json_data = dict()
+    for key, value in data.items():
+        if isinstance(value, list): # for lists
+            value = [ jsonify(item) if isinstance(item, dict) else item for item in value ]
+        if isinstance(value, dict): # for nested lists
+            value = jsonify(value)
+        if isinstance(key, int): # if key is integer: > to string
+            key = str(key)
+        if type(value).__module__=='numpy': # if value is numpy.*: > to python list
+            value = value.tolist()
+        json_data[key] = value
+    return json_data
+
+for res in result:
+    result_json.append(jsonify(res))
+
+with open(DETAILS_DIR + '/results.json', 'w') as outfile:
+    json.dump(result_json, outfile)
+
+output_images = get_recipe_config().get('save_image')
```

Comment:
to improve: get all the recipe config at one place, better to follow/update later

Reply:
done